mergin/client.py

```diff
@@ -618,3 +626,67 @@ def get_projects_by_names(self, projects):
 
         resp = self.post("/v1/project/by_names", {"projects": projects}, {"Content-Type": "application/json"})
         return json.load(resp)
+
+    def download_file(self, project_dir, file_path, output_filename, version=None):
+        """
+        Download project file at specified version. Get the latest if no version specified.
+
+        :param project_dir: project local directory
+        :type project_dir: String
+        :param file_path: relative path of file to download in the project directory
+        :type file_path: String
+        :param output_filename: full destination path for saving the downloaded file
+        :type output_filename: String
+        :param version: optional version tag for downloaded file
+        :type version: String
+        """
+        try:
+            mp = MerginProject(project_dir)
+        except InvalidProject as e:
+            err = f"Couldn't create Mergin project for {project_dir}: {repr(e)}"
+            self.log.error(err)
+            raise ClientError(err)
+
+        project_path = mp.metadata["name"]
+        ver_info = f"at version {version}" if version is not None else "at latest version"
+        self.log.info(f"Getting {file_path} from {project_path} {ver_info}")
+        job = download_file_async(self, project_path, file_path, output_filename, version=version)
+        if job is None:
+            return
+        pull_project_wait(job)
+        return download_file_finalize(job)
+
+    def get_file_diff(self, project_dir, file_path, output_diff, version_from, version_to):
+        """ Create concatenated diff for project file diffs between versions version_from and version_to.
+
+        :param project_dir: project local directory
+        :type project_dir: String
+        :param file_path: relative path of file to download in the project directory
+        :type file_path: String
+        :param output_diff: full destination path for concatenated diff file
+        :type output_diff: String
+        :param version_from: starting project version tag for getting diff, for example 'v3'
+        :type version_from: String
+        :param version_to: ending project version tag for getting diff
+        :type version_to: String
+        """
+        try:
+            mp = MerginProject(project_dir)
+        except InvalidProject as e:
+            err = f"Couldn't create Mergin project for {project_dir}: {repr(e)}"
+            self.log.error(err)
+            raise ClientError(err)
+
+        project_path = mp.metadata["name"]
+        self.log.info(f"Getting diffs for file {file_path} of {project_path}")
+        file_history = self.project_file_history_info(project_path, file_path)
+        versions_to_fetch = get_versions_with_file_changes(
+            self, project_path, file_path, version_from=version_from, version_to=version_to, file_history=file_history
+        )
```

Comment:
like in the comment above, I would move all the above to download_diffs_async so that all the logic is in one place